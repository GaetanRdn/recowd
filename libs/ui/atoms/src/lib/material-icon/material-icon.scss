@mixin rc-material-icon($config) {
  $colors: map-get($config, colors);
  $primary: map-get($colors, primary);

  .material-icons,
  .material-icons-outlined,
  .material-icons-round,
  .material-icons-sharp,
  .material-icons-two-tone {
    color: map-get($primary, color);
  }
}
