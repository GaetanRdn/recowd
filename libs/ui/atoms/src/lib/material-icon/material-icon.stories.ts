import {
  componentWrapperDecorator,
  Meta,
  moduleMetadata,
  Story,
} from '@storybook/angular';
import {
  MaterialIconComponent,
  MaterialIconModule,
} from './material-icon.component';
import { CommonModule } from '@angular/common';
import materialIcons from './material-icon.json';
import { HttpClientModule } from '@angular/common/http';
import { BADGE } from '@geometricpanda/storybook-addon-badges';

export default {
  title: 'atoms/material-icon',
  component: MaterialIconComponent,
  parameters: {
    badges: [BADGE.STABLE],
    jest: 'material-icon.component',
  },
  decorators: [
    moduleMetadata({
      imports: [MaterialIconModule, CommonModule, HttpClientModule],
    }),
    componentWrapperDecorator(
      (story: string) =>
        `<div style="display: grid; gap: 2rem; grid-template-columns: repeat(auto-fill, minmax(11.2rem, 1fr))">${story}</div>`
    ),
  ],
  argTypes: {
    type: { control: false },
  },
} as Meta<MaterialIconComponent>;

const template: Story<MaterialIconComponent> = (
  args: MaterialIconComponent
) => ({
  props: {
    type: args.type,
    materialIcons: materialIcons.sort((i1, i2) =>
      i1.popularity > i2.popularity ? -1 : i1.popularity < i2.popularity ? 1 : 0
    ),
  },
  template: `
    <div *ngFor="let icon of materialIcons" style="display: flex; flex-flow: column; align-items: center; justify-content: space-between">
    <rc-material-icon [type]="type">{{icon.name}}</rc-material-icon>
    <span style="font-size: 1.2rem; color: white; line-height: 1.6rem; font-family: 'Source Code Pro',serif">{{ icon.name }}</span>
</div>
`,
});

export const Default = template.bind({});

export const Outlined = template.bind({});
Outlined.args = {
  type: 'outlined',
};

export const TwoTone = template.bind({});
TwoTone.args = {
  type: 'two-tone',
};

export const Sharp = template.bind({});
Sharp.args = {
  type: 'sharp',
};

export const Rounded = template.bind({});
Rounded.args = {
  type: 'rounded',
};
