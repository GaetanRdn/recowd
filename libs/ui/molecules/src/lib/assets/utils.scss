@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
