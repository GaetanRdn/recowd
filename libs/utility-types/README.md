# utility-types

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test utility-types` to execute the unit tests.
