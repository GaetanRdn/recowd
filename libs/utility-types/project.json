{
  "projectType": "library",
  "sourceRoot": "libs/utility-types/src",
  "prefix": "rc",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:ng-packagr-lite",
      "outputs": ["dist/libs/utility-types"],
      "options": {
        "project": "libs/utility-types/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/utility-types/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/utility-types/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/utility-types"],
      "options": {
        "jestConfig": "libs/utility-types/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/utility-types/src/**/*.ts",
          "libs/utility-types/src/**/*.html"
        ]
      }
    }
  },
  "tags": ["utils"]
}
