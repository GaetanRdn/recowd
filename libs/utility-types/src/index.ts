export * from './lib/boolean-input';
export * from './lib/coerce-boolean-input.decorator';
export * from './lib/coerce-string-input.decorator';
export * from './lib/common';
export * from './lib/ng-changes';
export * from './lib/required.decorator';
export * from './lib/typed-control-value-accessor';
