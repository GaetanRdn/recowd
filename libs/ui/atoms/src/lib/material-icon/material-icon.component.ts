import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { CoerceString } from '@recowd/utility-types';

@Component({
  selector: 'rc-material-icon',
  standalone: true,
  host: {
    '[class.material-icons]': 'type === "filled"',
    '[class.material-icons-outlined]': 'type === "outlined"',
    '[class.material-icons-two-tone]': 'type === "two-tone"',
    '[class.material-icons-sharp]': 'type === "sharp"',
    '[class.material-icons-round]': 'type === "rounded"',
  },
  template: ` <ng-content></ng-content>`,
  styleUrls: ['./material-icon.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class MaterialIconComponent {
  @Input() @CoerceString('filled') public type:
    | 'filled'
    | 'outlined'
    | 'two-tone'
    | 'sharp'
    | 'rounded' = 'filled';
}
