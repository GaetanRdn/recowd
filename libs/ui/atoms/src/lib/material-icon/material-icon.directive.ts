import { Directive, Input } from '@angular/core';
import { CoerceString } from '@recowd/utility-types';

@Directive({
  selector: 'rc-material-icon, [rcMaterialIcon]',
  standalone: true,
  host: {
    '[class.material-icons]': 'type === "filled"',
    '[class.material-icons-outlined]': 'type === "outlined"',
    '[class.material-icons-two-tone]': 'type === "two-tone"',
    '[class.material-icons-sharp]': 'type === "sharp"',
    '[class.material-icons-round]': 'type === "rounded"',
  },
})
export class MaterialIconDirective {
  @Input() @CoerceString('filled') public type:
    | 'filled'
    | 'outlined'
    | 'two-tone'
    | 'sharp'
    | 'rounded' = 'filled';
}
