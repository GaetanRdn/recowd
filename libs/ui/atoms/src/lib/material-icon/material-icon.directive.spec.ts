import { Component } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';
import { MaterialIconDirective } from './material-icon.directive';
import { TemplateLookup } from '@recowd/test/utils';
import { Nullable } from '@recowd/utility-types';

describe('MaterialIconDirective', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() =>
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [MaterialIconDirective],
    }).compileComponents()));

  beforeEach(() => (templateLookup = new TemplateLookup(HostComponent)));

  test.each(['home', 'pin_push'])('should create icon %s', (icon: string) => {
    // WHEN
    templateLookup.hostComponent.icon = icon;
    templateLookup.detectChanges();

    // THEN
    expect(templateLookup.firstChildElement).toMatchSnapshot();
  });

  test.each([
    'filled',
    'outlined',
    'two-tone',
    'sharp',
  ] as MaterialIconDirective['type'][])(
    'should be of type %s',
    (type: MaterialIconDirective['type']) => {
      // WHEN
      templateLookup.hostComponent.icon = 'home';
      templateLookup.hostComponent.type = type;
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );

  test.each([undefined, null])(
    'should be filled when type is %s',
    (type?: Nullable<MaterialIconDirective['type']>) => {
      // WHEN
      templateLookup.hostComponent.icon = 'home';
      templateLookup.hostComponent.type = type;
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }
  );
});

@Component({
  template: ` <rc-material-icon [type]="type">{{ icon }}</rc-material-icon>`,
})
class HostComponent {
  public icon!: string;
  public type?: Nullable<MaterialIconDirective['type']>;
}
