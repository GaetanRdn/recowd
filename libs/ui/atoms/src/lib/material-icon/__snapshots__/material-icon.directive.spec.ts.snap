// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialIconDirective should be filled when type is null 1`] = `
<rc-material-icon
  class="material-icons"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should be filled when type is undefined 1`] = `
<rc-material-icon
  class="material-icons"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should be of type filled 1`] = `
<rc-material-icon
  class="material-icons"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should be of type outlined 1`] = `
<rc-material-icon
  class="material-icons-outlined"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should be of type sharp 1`] = `
<rc-material-icon
  class="material-icons-sharp"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should be of type two-tone 1`] = `
<rc-material-icon
  class="material-icons-two-tone"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should create icon home 1`] = `
<rc-material-icon
  class="material-icons"
>
  home
</rc-material-icon>
`;

exports[`MaterialIconDirective should create icon pin_push 1`] = `
<rc-material-icon
  class="material-icons"
>
  pin_push
</rc-material-icon>
`;
